src/app/components/shell/header/header.js

```diff
@@ -338,7 +339,10 @@ class Header extends BaseView {
             let loginItem = this.el.querySelector('.meta-nav .container .login a'),
                 userInfo = data[0];
 
-            if (userInfo && userInfo.username) {
+            if (!userInfo || userInfo.username === '') {
+                this.meta.loggedIn = false;
```

Comment:
This seems like it shouldn't work right, but maybe this is something I don't understand properly. The templateHelpers are used when the template is rendered, and then after rendering you're (possibly) changing the values used in templateHelpers. Is the templating system smart enough to re-render when its helper values change?